rfd/0184-agent-auto-updates.md

```diff
@@ -0,0 +1,1640 @@
+
+### Details - Kubernetes Agents
+
+The Kubernetes agent updater will be updated for compatibility with the new scheduling system.
+
+This means that it will stop reading update windows using the authenticated connection to the proxy, and instead update when indicated by the `/v1/webapi/find` endpoint.
+
+Rollbacks for the Kubernetes updater, as well as packaging changes to improve UX and compatibility, will be covered in a future RFD.
+
+## Migration
+
+The existing update system will remain in-place until the old auto-updater is fully deprecated.
+
+Both update systems can co-exist on the same machine.
+The old auto-updater will update the system package, which will not affect the `teleport-update`-managed installation. 
+
+Eventually, the `cluster_maintenance_config` resource and `teleport-ent-upgrader` package will be deprecated.
+
+## Security
+
+The initial version of automatic updates will rely on TLS to establish
+connection authenticity to the Teleport download server. The authenticity of
+assets served from the download server is out of scope for this RFD. Cluster
+administrators concerned with the authenticity of assets served from the
+download server can use self-managed updates with system package managers which
+are signed.
+
+The Update Framework (TUF) will be used to implement secure updates in the future.
+
+Anyone who possesses an updater UUID can determine when that host is scheduled to update by repeatedly querying the public `/v1/webapi/find` endpoint.
+It is not possible to discover the current version of that host, only the designated update window.
+
+## Logging
+
+All installation steps will be logged locally, such that they are viewable with `journalctl`.
+Care will be taken to ensure that updater logs are sharable with Teleport Support for debugging and auditing purposes.
+
+When TUF is added, that events related to supply chain security may be sent to the Teleport cluster via the Teleport Agent.
+
+## Alternatives
+
+### `teleport update` Subcommand
+
+`teleport-update` is intended to be a minimal binary, with few dependencies, that is used to bootstrap initial Teleport agent installations.
+It may be baked into AMIs or containers.
+
+If the entirely `teleport` binary were used instead, security scanners would match vulnerabilities all Teleport dependencies, so customers would have to handle rebuilding artifacts (e.g., AMIs) more often.
+Deploying these updates is often more disruptive than a soft restart of the agent triggered by the auto-updater.
+
+`teleport-update` will also handle `tbot` updates in the future, and it would be undesirable to distribute `teleport` with `tbot` just to enable automated updates.
+
+Finally, `teleport-update`'s API contract with the cluster must remain stable to ensure that outdated agent installations can always be recovered.
+The first version of `teleport-update` will need to work with Teleport v14 and all future versions of Teleport.
```

Comment:
Even if we did have enough time, and managed to get this in, it would only be compatible with a very small percentage of v14 releases to date.

Probably easier to just support latest v15 and up.

Reply:
Resolved during call: we can backport to v14 if necessary to migrate Cloud customers off v14.